playbooks/remove_node.yml

```diff
@@ -27,6 +27,10 @@
   hosts: "{{ node | default('kube_node') }}"
   gather_facts: false
   environment: "{{ proxy_disable_env }}"
+  pre_tasks:
+    - name: Gather information about installed services
+      service_facts:
+      when: reset_nodes | default(True) | bool
```

Comment:
1. Could you explain why this is needed ? This is not obvious from the code / commits messages alone.
2. What's the reason to use a pre_tasks ? This is harder to do inside the roles themselves ?

Reply:
IIUC this is to mimic the behavior that we have in reset.yml that does exactly this too (but I am not entirely sure too how this is used after that)

Reply:
> Could you explain why this is needed ? This is not obvious from the code / commits messages alone.

I opened PR and forgot to describe the problem in detail. 😅

I have compared the `remove_node.yml` and `reset.yml` diffs, and before adding these changes, `ansible_facts.services["containerd.service"]` would fail to check for triggering the container stop and delete behaviors.

Reference: https://github.com/kubernetes-sigs/kubespray/blob/616e4b40dbbb5ffa637d31d78ba2253645782b87/roles/reset/tasks/main.yml#L67-L74

> What's the reason to use a pre_tasks ? This is harder to do inside the roles themselves ?

I see that `reset.yml` is written in pre_task, as long as services are checked before reset.
It's OK if you want to write them all in `facts.yml`.